src/tests/test_sphincsplus.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+* (C) 2023 Jack Lloyd
+*     2023 Fabian Albert, René Meusel, Amos Treiber - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "test_rng.h"
+#include "tests.h"
+
+#if(defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHA2) || defined(BOTAN_HAS_SPHINCS_PLUS_WITH_SHAKE)) && defined(BOTAN_HAS_AES)
+
+   #include <botan/hash.h>
+   #include <botan/hex.h>
+
+   #include <botan/assert.h>
+   #include <botan/hash.h>
+   #include <botan/pk_algs.h>
+   #include <botan/pubkey.h>
+   #include <botan/secmem.h>
+   #include <botan/sp_parameters.h>
+   #include <botan/sphincsplus.h>
+   #include <botan/internal/loadstor.h>
+   #include <botan/internal/sp_hash.h>
+
+   #include "test_pubkey.h"
+
+namespace Botan_Tests {
+
+/**
+ * Test all implemented SPHINCS+ instances using the data of the KAT files
+ * generated by the reference implementation (Round 3.1
+ *  (https://github.com/sphincs/sphincsplus/))
+ */
+class SPHINCS_Plus_Test final : public Text_Based_Test {
+   public:
+      SPHINCS_Plus_Test() : Text_Based_Test("pubkey/sphincsplus.vec", "SphincsParameterSet,seed,pk,sk,msg,HashSig") {}
+
+      bool skip_this_test(const std::string&, const VarMap& vars) override {
+         [[maybe_unused]] auto params = Botan::Sphincs_Parameters::create(vars.get_req_str("SphincsParameterSet"));
```

Comment:
This should never really be unused because we know that at least one of SHA-2 or SHAKE is enabled, right?

Reply:
The lines below are not `#if defined(...)` but `#if not defined(...)`. If both hash functions are enabled, `params` is unused.